Tests/UnitTests/Purchasing/Purchases/BasePurchasesTests.swift

```diff
@@ -87,14 +93,44 @@ class BasePurchasesTests: TestCase {
         // See `addTeardownBlock` docs:
         // - These run *before* `tearDown`.
         // - They run in LIFO order.
-
         self.addTeardownBlock { [weak purchases = self.purchases] in
-            expect(purchases).to(beNil(), description: "Purchases has leaked")
+            expect(purchases).toEventually(beNil(), description: "Purchases has leaked")
+        }
+        self.addTeardownBlock { [weak orchestrator = self.purchasesOrchestrator] in
+            expect(orchestrator).toEventually(beNil(), description: "PurchasesOrchestrator has leaked")
+        }
+        self.addTeardownBlock { [weak deviceCache = self.deviceCache] in
+            expect(deviceCache).toEventually(beNil(), description: "DeviceCache has leaked: \(self)")
         }
 
         self.addTeardownBlock {
             Purchases.clearSingleton()
 
+            self.mockOperationDispatcher = nil
+            self.paymentQueueWrapper = nil
+            self.requestFetcher = nil
+            self.receiptFetcher = nil
+            self.mockProductsManager = nil
+            self.mockIntroEligibilityCalculator = nil
+            self.mockTransactionsManager = nil
+            self.backend = nil
+            self.attributionFetcher = nil
+            self.purchasesDelegate.makeDeferredPurchase = nil
+            self.purchasesDelegate = nil
+            self.storeKit1Wrapper.delegate = nil
+            self.storeKit1Wrapper = nil
+            self.systemInfo = nil
+            self.notificationCenter = nil
+            self.subscriberAttributesManager = nil
+            self.trialOrIntroPriceEligibilityChecker = nil
+            self.attributionPoster = nil
+            self.attribution = nil
+            self.customerInfoManager = nil
+            self.identityManager = nil
+            self.mockOfferingsManager = nil
+            self.mockManageSubsHelper = nil
+            self.mockBeginRefundRequestHelper = nil
+            self.purchasesOrchestrator = nil
```

Comment:
This is a lot, but it's been the only way to remove one by one the references at the end of the tests to see what's leaking. And there's still some retain cycle somewhere!

Reply:
So far the ones I've found are only in tests, but `DeviceCache` is still leaking. I need to spend a bit more time to find the last cause, which might be in the main SDK.